app/controllers/application_controller.rb

```diff
@@ -69,7 +69,13 @@ def render_unauthorized
   end
 
   def saml_settings
-    SAML::SettingsService.instance.saml_settings
+    settings = SAML::SettingsService.instance.saml_settings
+    # TODO: 'level' should be its own class with proper validation
+    level = Integer(params[:level]) unless params[:level].nil?
+    settings.authn_context = LOA::MAPPING.invert[level || 1]
+    settings
+  rescue ArgumentError
+    render json: {message: "Invalid Level."}, status: :bad_request
```

Comment:
The issue is that this method is rendering as well as the method that called this. You should re-raise the error to avoid control being given back to the caller. The global error handler in ApplicationController will catch it and render the appropriate JSON response. @saneshark can probably give you a better idea of which error class to construct and re-raise (or you can take a look at them in `lib/common/exceptions`